src/main/java/hudson/plugins/copyartifact/Copier.java

```diff
@@ -35,7 +35,7 @@
      * @param baseTargetDir Base target dir for upcoming file copy (the copy-artifact
      *   build step may later specify a deeper target dir)
      */
-    public abstract void init(Run src, AbstractBuild<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
+    public abstract void init(Run src, Run<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
```

Comment:
https://github.com/jenkinsci/jenkins/commit/5b6522808090ae675826a6033382f2361f7f270c was the original change.